components/org.wso2.carbon.identity.outbound.provisioning.connector.office365/src/main/java/org/wso2/carbon/identity/outbound/provisioning/connector/office365/Office365ProvisioningConnector.java

```diff
@@ -0,0 +1,420 @@
+     *
+     * @param provisioningEntity user to be provisioned
+     * @return string id for the provisioned user
+     * @throws IdentityProvisioningException if the user can not be created in the Azure AD
+     */
+    protected String createUser(ProvisioningEntity provisioningEntity) throws IdentityProvisioningException {
+
+        String provisionedId = null;
+
+        try (CloseableHttpClient httpclient = HttpClientBuilder.create().useSystemProperties().build()) {
+
+            JSONObject user = buildUserAsJson(provisioningEntity);
+            HttpPost post = new HttpPost(Office365ConnectorConstants.OFFICE365_USER_ENDPOINT);
+            setAuthorizationHeader(post);
+
+            StringEntity requestBody = new StringEntity(user.toString());
+            requestBody.setContentType(Office365ConnectorConstants.CONTENT_TYPE_APPLICATION_JSON);
+            post.setEntity(requestBody);
+            post.setHeader(Office365ConnectorConstants.CONTENT_TYPE, Office365ConnectorConstants
+                    .CONTENT_TYPE_APPLICATION_JSON);
+
+            try (CloseableHttpResponse response = httpclient.execute(post)) {
+
+                JSONObject jsonResponse = new JSONObject(new JSONTokener(new InputStreamReader(
+                        response.getEntity().getContent())));
+                if (response.getStatusLine().getStatusCode() == HttpStatus.SC_CREATED) {
+                    provisionedId = jsonResponse.getString("id");
+
+                    if (log.isDebugEnabled()) {
+                        log.debug("Successfully created an user in the Azure Active Directory. Server responds with " +
+                                jsonResponse.toString());
+                    }
+                } else {
+                    String errorMessage = jsonResponse.getJSONObject("error").getString("message");
+                    log.error("Received response status code: " + response.getStatusLine().getStatusCode() + " "
+                            + response.getStatusLine().getReasonPhrase() + " with the message '" + errorMessage +
+                            "' while creating the user " + user.getString(Office365ConnectorConstants.OFFICE365_UPN) +
+                            " in the Azure Active Directory.");
+
+                    if (log.isDebugEnabled()) {
+                        log.debug("The response received from server : " + jsonResponse.toString());
+                    }
+                }
+            } catch (IOException | JSONException e) {
+                throw new IdentityProvisioningException("Error while executing the create operation in user " +
+                        "provisioning", e);
+            }
+
+            if (log.isDebugEnabled()) {
+                log.debug("Returning provisioned user's ID: " + provisionedId);
+            }
+        } catch (IOException e) {
+            log.error("Error while closing HttpClient.");
+        }
+        return provisionedId;
+    }
+
+    protected void updateUser() {
```

Comment:
Update user method without any arguments seems incorrect. Is this the expected method signature?
Also would be better to track the TODO with a git issue.

Reply:
Created a new PR, passing the ProvisioningEntity to this update user method [1]

Created a git issue to track the TODO [2]

[1] https://github.com/wso2-extensions/identity-office365/pull/5
[2] https://github.com/wso2-extensions/identity-office365/issues/4